### ISSUE TITLE
Postgresql "smart" shutdown hangs indefinitely on leader change

### ISSUE THREAD

**Issue by sjagoe** [kubedb/project#768]
When leadership change occurs and the database is in use, shutdown fails because the application has open connections to the database. This leads to a state where the application and database are in a degraded state; no new connections can be accepted, but the database isn't restarting for leadership change.

```
I0605 17:40:17.270248       1 leaderelection.go:263] failed to renew lease namespace/postgresql-leader-lock: failed to tryAcquireOrRenew context deadline exceeded
I0605 17:40:17.270270       1 log.go:172] Lost leadership, initiating a restart to correctly signal the database
I0605 17:40:17.270274       1 log.go:172] Terminating postgres server and operator
I0605 17:40:17.270276       1 log.go:172] Postgres is running, sending SIGTERM
I0605 17:40:17.270286       1 log.go:172] Waiting for postgres to terminate
2020-06-05 17:40:17.270 UTC [26] LOG:  received smart shutdown request
2020-06-05 17:40:17.271 UTC [26] LOG:  worker process: logical replication launcher (PID 11868) exited with exit code 1
2020-06-05 17:40:19.799 UTC [14166] FATAL:  the database system is shutting down
2020-06-05 17:40:24.816 UTC [14174] FATAL:  the database system is shutting down
2020-06-05 17:40:29.823 UTC [14181] FATAL:  the database system is shutting down
2020-06-05 17:40:34.807 UTC [14188] FATAL:  the database system is shutting down
...
```

It would be nice if, after some timeout (e.g. like `gracePeriodSeconds`) with the database not exiting, pg-leader-election would send a `SIGINT` to postgres to initiate a fast shutdown.

**Comment by sjagoe** [comment]
The workarounds I'm testing for this are:
- increasing the `leaderElection` duration and deadline periods
- adding a liveness probe to detect this condition, forcing kubelet to restart these containers

**Comment by tamalsaha** [comment]
This issue has been resolved in KubeDB release v2021.03.17 . Please open a new bug if you still see this issue.
